mods/cnc/maps/gdi09/gdi09.lua

```diff
@@ -0,0 +1,216 @@
+		count = 0
+		local actors = enemy.GetActorsByType(type)
+		Utils.Do(actors, function(actor)
+			if actor.IsIdle and count < amount then
+				SetAttackWaypoints(actor, team.waypoints)
+				IdleHunt(actor)
+				count = count + 1
+			end
+		end)
+	end
+end
+
+SetAttackWaypoints = function(actor, waypoints)
+	if not actor.IsDead then
+		Utils.Do(waypoints, function(waypoint)
+			actor.AttackMove(waypoint.Location)
+		end)
+	end
+end
+
+StartPatrols = function()
+	for k, team in pairs(Patrols) do
+		local group = 1
+		for type, amount in pairs(team.units) do
+			for i=1,amount,1 do
+				Reinforcements.Reinforce(enemy, { type }, { team.waypoints[team.initialWaypointPlacement[group]] }, 0, function(unit)
+					ReplenishPatrolUnit(unit, handofnod, team.waypoints, team.wait)
+				end)
+			end
+			group = group + 1
+		end
+	end
+	Patrols = nil
+end
+
+ReplenishPatrolUnit = function(unit, building, waypoints, waitatwaypoint)
+	unit.Patrol(waypoints, true, waitatwaypoint)
+	Trigger.OnKilled(unit, function()
+		local queueUnit = { unit = { unit.Type }, atbuilding = { building }, waypoints = waypoints }
+		PatrolProductionQueue[#PatrolProductionQueue + 1] = queueUnit
+	end)
+end
+
+SendGDIReinforcements = function()
+	Media.PlaySpeechNotification(player, "Reinforce")
+	Reinforcements.ReinforceWithTransport(player, InsertionHelicopterType, GDIHeliReinfUnits, { GDIHeliEntryNorth.Location, GDIHeliLZ.Location }, { GDIHeliLZ.Location + CVec.New(20, 0) })
+end
+
+SendGDIReinforcementChinook = function()
+	Reinforcements.ReinforceWithTransport(player, 'tran', nil, { GDIHeliEntryNorth.Location, GDIHeliLZ.Location })
+end
+
+MarkCompletedObjectiveBunkers = function()
+	if not player.IsObjectiveCompleted(gdiMainObjectiveBunkers) then
+		player.MarkCompletedObjective(gdiMainObjectiveBunkers)
+	end
+end
+
+SpawnGunboat = function()
+	Media.PlaySpeechNotification(player, "Reinforce")
+	Actor.Create("boat", true, { Owner = player, Facing = 0, Location = waypoint23.Location })
+end
+
+WorldLoaded = function()
+	player = Player.GetPlayer("GDI")
+	enemy = Player.GetPlayer("Nod")
+
+	Camera.Position = DefaultCameraPosition.CenterPosition
+	
+	StartStationaryGuards()
+	
+	StartAI(nodconyard)
+
+	Trigger.OnObjectiveAdded(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. string.lower(p.GetObjectiveType(id)) .. " objective")
+	end)
+
+	Trigger.OnObjectiveCompleted(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+	end)
+
+	Trigger.OnObjectiveFailed(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
+	end)
+
+	Trigger.OnPlayerWon(player, function()
+		Media.PlaySpeechNotification(enemy, "Win")
+	end)
+
+	Trigger.OnPlayerLost(player, function()
+		Media.PlaySpeechNotification(enemy, "Lose")
+	end)
+
+	gdiMainObjectiveBunkers = player.AddPrimaryObjective("Destroy the Nod bunkers to allow Carter's\nconvoy to pass through safely.")
```

Comment:
You should use `player.AddObjective` instead (see the Lua API page). Also: Global variables should start with an upper case letter (and I would write GDI all upper case anyway).

Reply:
Can't find a AddObjective Function in the [Lua API page](https://github.com/OpenRA/OpenRA/wiki/Lua-API) but the change works.

Reply:
The page is for the last release, and therefore won't list new, unreleased features. (Although the top of the page shows `{DEV_VERSION}` which is indeed a bug... Fwiw, https://github.com/OpenRA/OpenRA/wiki/Lua-API-(playtest) already shows this method.)

Reply:
Should I also change the `AddSecondaryObjective` to 
`AddObjective("description", "Secondary", false)`?